### PR TITLE
Add "merge activist"

### PR DESCRIPTION
**Pull request by samertm**
Add feature to merge one activist into another.

It works by hiding the activist you're merging and assigning all of their attendance to the "merged" user.

 - Add /activist/merge
 - Add `merged_activist_attendance` table so we can keep track of the merges that have been done so that they can be undone. The undo feature isn't implemented yet, we'll do that once we let people view hidden activists.

Some things I'd like feedback on:
 - Is there a better name than "mergedUserID" for the ID that we're merging into?
 - We don't have any checks that keep people from using the pre-merged user when taking attendance. Should we check that an activist hasn't been merged when we check attendance?

### CODE REVIEW

File: model/activist.go [hunk 1]
```diff
@@ -348,6 +348,106 @@ func HideUser(db *sqlx.DB, userID int) error {
...
+	}
+
+	// Let's do this the boring way.
+	var originalUserEventIDs []int
+	err = tx.Select(&originalUserEventIDs, `
+SELECT event_id
+FROM event_attendance
+WHERE
+  activist_id = ?
+`, originalUserID)
```
Comment (mleanos) [inline model/activist.go]: You might be able to simplify the logic that follow this query by changing it to exclude any events that the "merged" (target) user attended.

```sql
SELECT event_id
FROM event_attendance ea
WHERE
  activist_id = ? AND
  NOT EXISTS(
	SELECT ea2.event_id
    FROM event_attendance ea2
    WHERE ea2.activist_id = ? 
		AND ea.event_id = ea2.event_id)
```
Where the second query parameter is the "merged" (target) User.

File: model/activist.go [hunk 2]
```diff
@@ -348,6 +348,106 @@ func HideUser(db *sqlx.DB, userID int) error {
...
+			originalUserID)
+	}
+	for _, eventID := range originalUserEventIDs {
+		// If the merged user also went to the event, just
+		// delete the original user's attendance record.
+		// Otherwise, update the original user's attendance
+		// record so that the activist id is the merged user.
+		var mergedUserAttended int
+		err = tx.Get(&mergedUserAttended, `
+SELECT count(*) FROM event_attendance WHERE activist_id = ? AND event_id = ?
```
Comment (mleanos) [inline model/activist.go]: Related to my comment on the previous query excluding the "merged" User's duplicate events.. 

This wouldn't be needed, if we don't update or delete any of the "source" User's attendance records. By updating or deleting these records, it may complicate things when "undoing" a merge.

Perhaps we could handle this by just adding attendance records for the events (non-duplicate) found in the previous query. The "source" user is already being hidden, and we're tracking the merges. So whenever an undo merge feature is added, it would be just a couple simple operations.

1) Toggle the Activist's "hidden" field.
2) Remove the events that were merged into the "merged" Activist, based on the Activist ID that is getting "re-activated".

I usually try to avoid deletions when there's a possibility that they could be undone in the future.

File: frontend/event_list.js [hunk 3]
```diff
@@ -1,22 +1,5 @@
...
-      /IEMobile/i.test(window.navigator.userAgent) ||
-      /Windows Phone/i.test(window.navigator.userAgent) ||
-      /BlackBerry/i.test(window.navigator.userAgent) ||
-      /BB10/i.test(window.navigator.userAgent) ||
-      /Android.*Mobile/i.test(window.navigator.userAgent)) {
-    return false;
-  }
-  return true;
-}
+import {initActivistSelect} from 'chosen_utils';
```
Comment (mleanos) [inline frontend/event_list.js]: 👍 This is a good practice, and way to adhere to SOC.

File: model/activist.go [hunk 4]
```diff
@@ -348,6 +348,106 @@ func HideUser(db *sqlx.DB, userID int) error {
 	return nil
 }
 
+// Merge originalUserID into mergedUserID.
+//  - The original user is hidden
+//  - All of the original user's event attendance is updated to be the merged user.
+func MergeUser(db *sqlx.DB, originalUserID, mergedUserID int) error {
```
[the remark this answers is not in the document]
Comment (samertm) [reply model/activist.go]: done

File: model/activist.go (hunk 2 above)
Comment (samertm) [reply model/activist.go]: (One piece of context: hidden activists don't show up on the activist page, but they still get counted for attendance.)

I think our ideas are basically equivalent. You can completely reverse a merge with the info stored in merged_activist_attendance.

The one difference is that for the method you're suggesting, we'd need another field on activists (like `merged`) that indicates that the activist shouldn't show up on the activist page AND shouldn't be counted for attendance. And then when we count attendance, we'll need to join `event_attendance` and `activists` to only include unmerged activists.

File: model/activist.go (hunk 1 above)
Comment (samertm) [reply model/activist.go]: done

**Comment by samertm** [comment]
thank you for the review!

File: frontend/ActivistList.vue [hunk 5]
```diff
@@ -245,6 +323,18 @@ export default {
       // is shown.
       $(document.body).addClass('noscroll');
       this.disableConfirmButton = false;
+
+      if (this.currentModalName == "merge-activist-modal") {
+        // For some reason, even though this function is supposed to
```
Comment (cbailey20043) [inline frontend/ActivistList.vue]: yeah that's weird.

File: frontend/chosen_utils.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+  }
+
+  $.ajax({
+    url: "/activist_names/get",
+    method: "GET",
+    dataType: "json",
+    success: function(data) {
+      var activistNames = data.activist_names;
+
+      activistNames.unshift("");
```
Comment (cbailey20043) [inline frontend/chosen_utils.js]: Why is this needed exactly? Maybe add a brief comment explaining why

Comment (samertm) [reply frontend/chosen_utils.js]: done

File: model/activist.go (hunk 2 above)
Comment (mleanos) [reply model/activist.go]: So a "hidden" activist still shows up in the list for taking attendance? What's the point of setting the hidden field when merging then?
